src/test/java/de/tum/cit/aet/artemis/programming/web/ProgrammingExerciseResourceTest.java

```diff
@@ -0,0 +1,79 @@
+package de.tum.cit.aet.artemis.programming.web;
+
+import static org.assertj.core.api.Assertions.assertThat;
+
+import java.util.HashSet;
+import java.util.Set;
+
+import org.junit.jupiter.api.BeforeEach;
+import org.junit.jupiter.api.Test;
+import org.springframework.beans.factory.annotation.Autowired;
+import org.springframework.security.test.context.support.WithMockUser;
+
+import de.tum.cit.aet.artemis.core.domain.Course;
+import de.tum.cit.aet.artemis.core.test_repository.UserTestRepository;
+import de.tum.cit.aet.artemis.core.user.util.UserUtilService;
+import de.tum.cit.aet.artemis.core.util.CourseUtilService;
+import de.tum.cit.aet.artemis.exercise.participation.util.ParticipationUtilService;
+import de.tum.cit.aet.artemis.programming.domain.ProgrammingExercise;
+import de.tum.cit.aet.artemis.programming.repository.ProgrammingExerciseBuildConfigRepository;
+import de.tum.cit.aet.artemis.programming.util.ProgrammingExerciseUtilService;
+import de.tum.cit.aet.artemis.shared.base.AbstractSpringIntegrationIndependentTest;
+
+class ProgrammingExerciseResourceTest extends AbstractSpringIntegrationIndependentTest {
+
+    private static final String TEST_PREFIX = "programmingexerciseresource";
+
+    @Autowired
+    private ProgrammingExerciseResource programmingExerciseResource;
+
+    @Autowired
+    private ProgrammingExerciseBuildConfigRepository programmingExerciseBuildConfigRepository;
+
+    @Autowired
+    private UserUtilService userUtilService;
+
+    @Autowired
+    private CourseUtilService courseUtilService;
+
+    @Autowired
+    protected UserTestRepository userTestRepository;
+
+    @Autowired
+    protected ProgrammingExerciseUtilService programmingExerciseUtilService;
+
+    @Autowired
+    protected ParticipationUtilService participationUtilService;
+
+    protected Course course;
+
+    protected ProgrammingExercise programmingExercise;
+
+    @BeforeEach
+    void setup() {
+        String studentParticipationGroupName = TEST_PREFIX + "studentParticipationGroup";
+
+        userUtilService.addUsers(TEST_PREFIX, 1, 0, 0, 0);
+        var student1 = userUtilService.getUserByLogin(TEST_PREFIX + "student1");
+        Set<String> student1Groups = new HashSet<>(student1.getGroups());
+        student1Groups.add(studentParticipationGroupName);
+        student1.setGroups(student1Groups);
+        userTestRepository.save(student1);
+
+        course = programmingExerciseUtilService.addCourseWithOneProgrammingExercise();
+        course.setStudentGroupName(studentParticipationGroupName);
+        course = courseRepository.save(course);
+
+        programmingExercise = exerciseUtilService.getFirstExerciseWithType(course, ProgrammingExercise.class);
+        participationUtilService.addStudentParticipationForProgrammingExercise(programmingExercise, TEST_PREFIX + "student1");
+    }
+
+    @Test
+    @WithMockUser(username = TEST_PREFIX + "student1", roles = "STUDENT")
```

Comment:
the `@EnforceAtLeastStudentInExercise` annotation in getBuildConfig needs the user to have the role "USER" which should in theory always be the case, but for some reason isn't here. This should fix the test:
```suggestion
    @WithMockUser(username = TEST_PREFIX + "student1", roles = {"USER", "STUDENT"})
```

Reply:
Awesome, thanks!! 🥰